controllers/machine_controller_phases_test.go

```diff
@@ -517,6 +517,7 @@ var _ = Describe("Reconcile Machine Phases", func() {
 
 		// Set Deletion Timestamp.
 		machine.SetDeletionTimestamp(&deletionTimestamp)
+		machine.Finalizers = append(machine.Finalizers, "test")
```

Comment:
Also here, what is the rational of this change

Reply:
Left a comment in the PR, this is to avoid the fake client to delete the object immediately, they now support finalizers.